packages/wallets/src/routes/Router.tsx

```diff
@@ -6,43 +6,67 @@ import { CashierModalRoute } from './CashierModalRoute';
 import { CompareAccountsRoute } from './CompareAccountsRoute';
 import { WalletsListingRoute } from './WalletsListingRoute';
 
-const prefix = '/wallets';
+const walletsPrefix = '/wallets';
 
-type TRoutes = `${typeof prefix}${
+type TWalletsRoute =
     | ''
     | '/cashier'
     | '/cashier/deposit'
+    | '/cashier/on-ramp'
     | '/cashier/reset-balance'
     | '/cashier/transactions'
     | '/cashier/transfer'
     | '/cashier/withdraw'
-    | '/compare-account'}`;
+    | '/compare-accounts';
+
+export type TRoute = '/endpoint' | `${typeof walletsPrefix}${TWalletsRoute}`;
+
+// wallets routes which have their states
+interface WalletsRouteState {
+    '/cashier/transactions': { showPending?: boolean; transactionType?: 'deposit' | 'withdrawal' };
+    '/cashier/transfer': { toAccountLoginId: string };
+}
+
+type TStatefulRoute = TRoute & `${typeof walletsPrefix}${keyof WalletsRouteState}`;
```

Comment:
```suggestion
type TStatefulRoute = `${typeof walletsPrefix}${keyof WalletsRouteState}`;
```
Shouldnt statefulRoute only this?

Reply:
I intended this to be like "TRoutes, that are also among the listed among the WalletsRouteStates"
Removing `TRoute &` will not change the type, but it'll make it easier to make mistakes later on (I think), because `TStatefulRoute` won't be guaranteed a sub type of `TRoutes` because nothing stops the user to add non-`TRoutes`-matching routes to `WalletsRouteStates`, really hope this makes sense